x/mongo/driver/topology/topology_test.go

```diff
@@ -217,77 +212,82 @@ func TestServerSelection(t *testing.T) {
 		want := ServerSelectionError{Wrapped: context.Canceled, Desc: desc}
 		assert.Equal(t, err, want, "Incorrect error received. got %v; want %v", err, want)
 	})
-	t.Run("Timeout", func(t *testing.T) {
-		desc := description.Topology{
-			Servers: []description.Server{
-				{Addr: address.Address("one"), Kind: description.Standalone},
-				{Addr: address.Address("two"), Kind: description.Standalone},
-				{Addr: address.Address("three"), Kind: description.Standalone},
-			},
-		}
-		topo, err := New(nil)
-		noerr(t, err)
-		subCh := make(chan description.Topology, 1)
-		subCh <- desc
-		resp := make(chan error)
-		timeout := make(chan time.Time)
-		go func() {
-			state := newServerSelectionState(selectNone, timeout)
-			_, err := topo.selectServerFromSubscription(context.Background(), subCh, state)
-			resp <- err
-		}()
-
-		select {
-		case err := <-resp:
-			t.Errorf("Received error from server selection too soon: %v", err)
-		case timeout <- time.Now():
-		}
-
-		select {
-		case err = <-resp:
-		case <-time.After(100 * time.Millisecond):
-			t.Errorf("Timed out while trying to retrieve selected servers")
-		}
-
-		if err == nil {
-			t.Fatalf("did not receive error from server selection")
-		}
-	})
-	t.Run("Error", func(t *testing.T) {
-		desc := description.Topology{
-			Servers: []description.Server{
-				{Addr: address.Address("one"), Kind: description.Standalone},
-				{Addr: address.Address("two"), Kind: description.Standalone},
-				{Addr: address.Address("three"), Kind: description.Standalone},
-			},
-		}
-		topo, err := New(nil)
-		noerr(t, err)
-		subCh := make(chan description.Topology, 1)
-		subCh <- desc
-		resp := make(chan error)
-		timeout := make(chan time.Time)
-		go func() {
-			state := newServerSelectionState(selectError, timeout)
-			_, err := topo.selectServerFromSubscription(context.Background(), subCh, state)
-			resp <- err
-		}()
-
-		select {
-		case err = <-resp:
-		case <-time.After(100 * time.Millisecond):
-			t.Errorf("Timed out while trying to retrieve selected servers")
-		}
-
-		if err == nil {
-			t.Fatalf("did not receive error from server selection")
-		}
-	})
+	// TODO(GODRIVER-2348): Since we use the context timeout now, does this test
```

Comment:
Q:  Now that we use a context and not a ticker to timeout server selection, are these and subsequent tests still needed?